front_camera/front_camera/color_rectangle_detector.py

```diff
@@ -0,0 +1,91 @@
+"""
+動画中の背景下部の指定色検出クラス
+@author: bizyutyu YKhm20020 CHIHAYATAKU
+"""
+import cv2
+import os
+
+class ColorRectangleDetector:
+    # 色のHSV範囲を定数として定義
+    COLOR_BOUNDS = {
+        'YELLOW': ((20, 160, 160), (30, 255, 255)),
+        'RED': [
+            ((0, 70, 50), (10, 255, 255)),     # 赤色の低色相側
+            ((173, 70, 50), (180, 255, 255))    # 赤色の高色相側
+        ]
+    }
+
+    def __init__(self, color):
+        """コンストラクタ
+        Args:
+            color (str): 検出する色の名前。COLOR_BOUNDSに定義された色から選択する。
+        """
+        # 対象の色の範囲を保持
+        if color not in self.COLOR_BOUNDS:
+            raise ValueError(f"{color}は定義されていません。定義されている色: {list(self.COLOR_BOUNDS.keys())}")
+
+        self.bounds = self.COLOR_BOUNDS[color]
+        self.color = color
+
+    def detect_rectangle(self, frame):
+        """対象色の矩形を検出する。
+        Args:
+            frame (numpy.ndarray): 処理対象の画像データ
+        """
```

Comment:
```suggestion
    def detect_rectangle(self, frame) -> float:
        """対象色の矩形を検出する。
        Args:
            frame (numpy.ndarray): 処理対象の画像データ
        Returns:
            float: 検出した矩形の面積
        """
```
None が返らないならこんな感じになりそう。書き方間違ってたらごめんなさい。

Reply:
れびゅーありがとうございます
コメント追加しました。
5a0abca20b19216d2f3b5cf31a94f03d6959b749